Source/ablastr/math/fft/WrapRocFFT.cpp

```diff
@@ -66,6 +66,69 @@ namespace ablastr::math::anyfft
         return fft_plan;
     }
 
+    FFTplan CreatePlanMany(int * real_size, amrex::Real * real_array,
+                           Complex * complex_array, const direction dir, const int dim,
+                           int howmany, int * inembed, int istride, int idist,
+                           int * onembed, int ostride, int odist)
+    {
+        FFTplan fft_plan;
+
+        const std::size_t lengths[] = {AMREX_D_DECL(std::size_t(real_size[0]),
+                                                    std::size_t(real_size[1]),
+                                                    std::size_t(real_size[2]))};
+
+        const std::size_t roc_istride[] = {AMREX_D_DECL( (dir == direction::R2C) ? std::size_t(istride * real_size[0]) : std::size_t(istride * (real_size[0]/2+1)),
+                                                          std::size_t(istride * real_size[1]),
+                                                          std::size_t(istride * real_size[2]))};
+
+        const std::size_t roc_ostride[] = {AMREX_D_DECL( (dir == direction::R2C) ? std::size_t(ostride * (real_size[0]/2+1)) : std::size_t(ostride * real_size[0]),
+                                                          std::size_t(ostride*real_size[1]),
+                                                          std::size_t(ostride*real_size[2]))};
+
+        rocfft_plan_description desc = nullptr;
+        rocfft_status result = rocfft_plan_description_set_data_layout(
+                                desc,
+                                (dir == direction::R2C) ? rocfft_array_type_real : rocfft_array_type_hermitian_interleaved,
+                                (dir == direction::R2C) ? rocfft_array_type_hermitian_interleaved : rocfft_array_type_real,
+                                (const size_t *) inembed,
+                                (const size_t *) onembed,
+                                dim,
+                                roc_istride,
+                                (const std::size_t)(idist),
+                                dim,
+                                roc_ostride,
+                                (const std::size_t)(odist));
+        assert_rocfft_status("rocfft_plan_description_set_data_layout", result);
+
+
+        // Initialize fft_plan.m_plan with the vendor fft plan.
+        result = rocfft_plan_create(&(fft_plan.m_plan),
+                                    rocfft_placement_notinplace,
+                                    (dir == direction::R2C)
+                                    ? rocfft_transform_type_real_forward
+                                    : rocfft_transform_type_real_inverse,
+#ifdef AMREX_USE_FLOAT
+                                    rocfft_precision_single,
+#else
+                                    rocfft_precision_double,
+#endif
+                                    dim, lengths,
+                                    howmany, // number of transforms,
+                                    desc);
```

Comment:
What happens if we don't pass a `desc` at all?

Reply:
No need of the plan description anymore.
However, the rocFFT wrapper is currently less flexible than that of FFTW and cuFFT.